internal/server/server.go

```diff
@@ -65,15 +66,27 @@ func getValidClaim(claims *jwt.Claims, input string) string {
 
 func (s *Server) handleRoot() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		// Extract token from cookie
-		cookie, err := r.Cookie(s.cookieName)
-		if err != nil {
-			logAndError(w, http.StatusUnauthorized, err, "error reading cookie")
-			return
+		// Get token
+		var token string
+
+		if s.headerName != "" {
+			token = r.Header.Get(s.headerName)
+			// replicate the cookie look up behavior for a missing header
```

Comment:
nit: the comment seems unnecessary.  The code feels clean and explains the intent.